letsencrypt/client/tests/proof_of_possession_test.py

```diff
@@ -0,0 +1,82 @@
+"""Tests for proof_of_possession.py"""
+import Crypto.PublicKey.RSA
+import os
+import pkg_resources
+import unittest
+
+import mock
+
+from letsencrypt.acme import challenges
+from letsencrypt.acme import jose
+from letsencrypt.client import achallenges
+from letsencrypt.client import proof_of_possession
+from letsencrypt.client.display import util as display_util
+
+
+BASE_PACKAGE = "letsencrypt.client.tests"
+CERT0_PATH = pkg_resources.resource_filename(
+    BASE_PACKAGE, os.path.join("testdata", "cert.pem"))
+CERT1_PATH = pkg_resources.resource_filename(
+    BASE_PACKAGE, os.path.join("testdata", "cert-san.pem"))
+CERT2_PATH = pkg_resources.resource_filename(
+    BASE_PACKAGE, os.path.join("testdata", "matching_cert.pem"))
+KEY_PATH = pkg_resources.resource_filename(
+    BASE_PACKAGE, os.path.join("testdata", "rsa512_key.pem"))
+KEY = Crypto.PublicKey.RSA.importKey(pkg_resources.resource_string(
+    BASE_PACKAGE, os.path.join('testdata', 'rsa512_key.pem'))).publickey()
+
+
+class ProofOfPossessionTest(unittest.TestCase):
+    def setUp(self):
+        self.installer = mock.MagicMock()
+        self.installer.get_all_certs_keys.return_value = zip(
+            [CERT0_PATH, CERT1_PATH, CERT2_PATH], 3 * [KEY_PATH], 3 * [None])
+        self.proof_of_pos = proof_of_possession.ProofOfPossession(
+            self.installer)
+
+        hints = challenges.ProofOfPossession.Hints(
+            jwk=jose.JWKRSA(key=KEY), cert_fingerprints=(),
+            certs=(), serial_numbers=(), subject_key_identifiers=(),
+            issuers=(), authorized_for=())
+        challenge = challenges.ProofOfPossession(
+            alg=jose.RS256, nonce='zczv4HMLVe_0kimJ25Juig', hints=hints)
+        self.achall = achallenges.ProofOfPossession(
```

Comment:
This works because challb is really just a proxied Challenge object, but in reality challb's are ChallengeBody  (defined in acme/messages2.py) objects which proxy Challenge objects... I realize this is confusing.  Just want to make sure that everything is clear and understood.

You probably made this mistake because I have just realized RecoveryToken in the master branch has this exact same problem (likely due to the recent v02 ACME protocol change).  (My bad)  I will fix this in the recovery_contact branch.